src/pages/settings/SettingsPage.tsx

```diff
@@ -70,6 +70,18 @@ function SettingsPage() {
     }
   };
 
+  const isShowFooter = useMemo(() => {
+    const currentPath = pathname.split('/').pop();
+    const item = items.find(x => x.path === currentPath);
+    return item?.useDefaultFooter ?? true;
+  }, [pathname]);
```

Comment:
Instead of adding the `useDefaultFooter` property to `items` and then doing find on it, the ones which don't need the footer can be in an array like `['user-management', 'api-keys']` and then you can just check if that array contains `pathname.split('/').pop()`.

Or better yet. Move the footer into its own component and use it inside each settings component wherever required and just keep the outlet here.